tensorflow-framework/src/main/java/org/tensorflow/framework/metrics/MeanRelativeError.java

```diff
@@ -0,0 +1,176 @@
+   *     will always produce the same random tensor for a given shape and data type.
+   * @param type the type for the variables and result
+   */
+  protected MeanRelativeError(Ops tf, double[] normalizer, long seed, Class<T> type) {
+    this(tf, null, cast(tf, tf.constant(normalizer), type), seed, type);
+  }
+
+  /**
+   * Creates a MeanRelativeError metric
+   *
+   * @param tf the TensorFlow Ops
+   * @param name the name of the metric. If null, name defaults to {@link Class#getSimpleName()}.
+   * @param normalizer The normalizer values with same shape as predictions.
+   * @param seed the seed for random number generation. An initializer created with a given seed
+   *     will always produce the same random tensor for a given shape and data type.
+   * @param type the type for the variables and result
+   */
+  protected MeanRelativeError(Ops tf, String name, double[] normalizer, long seed, Class<T> type) {
+    this(tf, name, cast(tf, tf.constant(normalizer), type), seed, type);
+  }
+
+  /**
+   * Creates a MeanRelativeError metric using {@link Class#getSimpleName()} as the name
+   *
+   * @param tf the TensorFlow Ops
+   * @param normalizer The normalizer values with same shape as predictions.
+   * @param seed the seed for random number generation. An initializer created with a given seed
+   *     will always produce the same random tensor for a given shape and data type.
+   * @param type the type for the variables and result
+   */
+  protected MeanRelativeError(Ops tf, Operand<T> normalizer, long seed, Class<T> type) {
+    this(tf, null, normalizer, seed, type);
+  }
+
+  /**
+   * Creates a MeanRelativeError metric
+   *
+   * @param tf the TensorFlow ops
+   * @param name the name for this metric. If null, name defaults to {@link Class#getSimpleName()}.
+   * @param normalizer The normalizer values with same shape as predictions.
+   * @param seed the seed for random number generation. An initializer created with a given seed
+   *     will always produce the same random tensor for a given shape and data type.
+   * @param type the type for the variables and result
+   */
+  protected MeanRelativeError(
+      Ops tf, String name, Operand<T> normalizer, long seed, Class<T> type) {
+    super(tf, name, seed, type);
+    this.normalizer = normalizer;
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public List<Op> updateStateList(
+      Operand<? extends TNumber> labels,
+      Operand<? extends TNumber> predictions,
+      Operand<? extends TNumber> sampleWeights) {
+    Operand<T> tLabels = cast(getTF(), labels, getResultType());
+    if (tLabels.shape().numDimensions() > 1) tLabels = getTF().shape.flatten(tLabels);
+
+    Operand<T> tPredictions = cast(getTF(), predictions, getResultType());
+    if (tPredictions.shape().numDimensions() > 1)
+      tPredictions = getTF().shape.flatten(tPredictions);
+
+    LossTuple<T> tuple = LossesHelper.squeezeOrExpandDimensions(getTF(), tLabels, tPredictions);
```

Comment:
This call doesn't make sense after we flattened both `tLabels` and `tPredictions`.

Reply:
I agree with you. It looks like I misinterpreted the Python` ragged_assert_compatible_and_get_flat_values `call.
I will fix the code and remove the `flatten` reshapes.